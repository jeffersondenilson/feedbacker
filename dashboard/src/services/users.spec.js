import mockAxios from 'axios'
import UserService from './users'

jest.mock('axios')

describe('UserService', () => {
  afterEach(() => {
    jest.clearAllMocks()
  })

  it('should return user info', async () => {
    const user = { name: 'user1', email: 'user1@gmail.com' }
    mockAxios.get.mockImplementationOnce(() => {
      return Promise.resolve({ data: user })
    })

    const response = await UserService(mockAxios).getMe()

    expect(response.data).toHaveProperty('name')
    expect(response.data).toHaveProperty('email')
    expect(response).toMatchSnapshot()
  })

  it('should return api key', async () => {
    const apiKey = 'a1p2i3k4e5y'
    mockAxios.get.mockImplementationOnce(() => {
      return Promise.resolve({ data: { apiKey } })
    })

    const response = await UserService(mockAxios).getMe()

    expect(response.data).toHaveProperty('apiKey')
    expect(response).toMatchSnapshot()
  })
})
