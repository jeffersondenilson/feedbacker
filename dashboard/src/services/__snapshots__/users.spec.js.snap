// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserService should return api key 1`] = `
Object {
  "data": Object {
    "apiKey": "a1p2i3k4e5y",
  },
}
`;

exports[`UserService should return user info 1`] = `
Object {
  "data": Object {
    "email": "user1@gmail.com",
    "name": "user1",
  },
}
`;
